### PR TITLE
Catch exception on call to iExec proxy

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

**Review by jbern0rd — APPROVED** [review]

File: src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java [hunk 1]
```diff
@@ -64,17 +66,22 @@ public static String uploadToIpfsWithIexecProxy(String taskId, String baseUrl, S
...
-            return response.getBody();
+        HttpStatus statusCode = null;
+        try {
+            ResponseEntity<String> response = new RestTemplate().postForEntity(baseUrl, request, String.class);
+            statusCode = response.getStatusCode();
+
+            if (statusCode.is2xxSuccessful()) {
+                return response.getBody();
+            }
+        } catch (HttpClientErrorException e) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java:77]: I think other exceptions can be thrown, such as `HttpServerErrorException` or `UnknownHttpStatusCodeException` or other (?)

Comment (mcornaton) [reply src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java]: Right, let's use `RestClientException` which is a common base class for all error thrown by `postForEntity`.

File: src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java [hunk 2]
```diff
@@ -64,17 +66,22 @@ public static String uploadToIpfsWithIexecProxy(String taskId, String baseUrl, S
...
-        ResponseEntity<String> response = new RestTemplate().postForEntity(baseUrl, request, String.class);
-
-        if (response.getStatusCode().is2xxSuccessful()) {
-            return response.getBody();
+        HttpStatus statusCode = null;
+        try {
+            ResponseEntity<String> response = new RestTemplate().postForEntity(baseUrl, request, String.class);
+            statusCode = response.getStatusCode();
+
+            if (statusCode.is2xxSuccessful()) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java:74]: status != null && is2xxSuccessful
(just in case)

Comment (mcornaton) [reply src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java:74]: `statusCode` could never be null here. When adding this check, IntellIj triggers a warning. If it's OK for you, let's NOT add this check :)

Comment (jeremyjams) [reply src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java:74]: Yes, thank you for checking!

**Review by jeremyjams — APPROVED** [review]

**Pull request merged**